internal/configv2/configv2.go

```diff
@@ -0,0 +1,406 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package configv2 implements loading of configuration V2 from Viper configuration.
+// The implementation relies on registered factories that allow creating
+// default configuration for each type of receiver/exporter/option.
+package configv2
+
+import (
+	"fmt"
+
+	"github.com/census-instrumentation/opencensus-service/internal/models"
+	"github.com/spf13/viper"
+)
+
+// These are errors that can be returned by Load(). Note that error codes are not part
+// of Load()'s public API, they are for internal unit testing only.
+type configErrorCode int
+
+const (
+	_ configErrorCode = iota // skip 0, start errors codes from 1
+	errUnknownReceiverType
+	errUnknownExporterType
+	errUnknownOptionType
+	errMissingReceiverType
+	errMissingExporterType
+	errMissingOptionType
+	errDuplicateReceiverName
+	errDuplicateExporterName
+	errDuplicateOptionName
+	errMissingPipelines
+	errMissingPipelineName
+	errPipelineMustHaveReceiver
+	errPipelineMustHaveExporter
+	errPipelineMustHaveOperations
+	errPipelineReceiverNotExists
+	errPipelineOptionNotExists
+	errPipelineExporterNotExists
+	errMetricPipelineCannotHaveOperations
+	errUnmarshalError
+)
+
+type configError struct {
+	msg  string          // human readable error message
+	code configErrorCode // internal error code
+}
+
```

Comment:
nit: things will break if that is not the case but add the `var _ error = (*configError)(nil)` to make it explicit.

Reply:
I am not sure I understand. Do you suggest to add this to ensure that `configError` satisfies the interface of `error`?

Reply:
yes

Reply:
Added.